external/vulkancts/modules/vulkan/api/vktApiMemoryRequirementInvarianceTests.cpp

```diff
@@ -180,12 +180,21 @@ ImageAllocator::ImageAllocator (deRandom& random, deBool dedicated, std::vector<
 	else
 		m_colorFormat = (VkFormat)optimalformats[deRandom_getUint32(&random) % optimalformats.size()];
 
-	int	widthAlignment	= (isYCbCr420Format(m_colorFormat) || isYCbCr422Format(m_colorFormat)) ? 2 : 1;
-	int	heightAlignment	= isYCbCr420Format(m_colorFormat) ? 2 : 1;
+	if (isPvrtc1Format(m_colorFormat))
+	{
+		m_size = tcu::IVec2(pow(2, deRandom_getUint32(&random) % 4),
```

Comment:
I'm getting `error: ‘pow’ was not declared in this scope`. We should use one of the de*pow versions.

I also think a comment should be added here to explain why we are doing this.

Reply:
Yes makes sense. Locally, on my Mac dePowFloat was not available, and I'm not sure from which header they'd come, as I am not familiar with this codebase. Can you point me to which header I should use? I'll fix the commit message and add the comment when I get home. Thanks.

Reply:
Sure, I will figure it out for you! We have many functions to align numbers to powers of two (flooring or ceiling numbers to them), and I was wondering if any of those could be used. I'll point you at them in a moment.

Reply:
Also, if present, it should be a dePowInteger. Would be best to avoid any explicit casts littering the code.

Reply:
We have `deFloatPow` with `#include "deMath.h"`, but as you said, it requires conversions, like `std::pow` (with `#include <cmath>`), which is also used elsewhere in code.

In terms of integer pow, we have `static int intPow (int base, int exp)` (recursive) in es31fTessellationTests.cpp, which could be moved to deMath.h, but at this point I'm not sure have a recursive function call is better than just converting from int to float and back...

Reply:
Another interesting one is `DE_INLINE deUint32 deSmallestGreaterOrEquallPowerOfTwoU32 (deUint32 a)` in deInt32.h which is quite efficient because it uses bit shifting.

Reply:
Perhaps it could be worth it to add a `deIntegerPow` function that just uses a simple loop to avoid float conversions entirely? I guess I could just use `deSmallestGreaterOrEquallPowerOfTwoU32` with `(deRandom_getUint32(&random) % 16)` to get a number divisible by 2, which would work in this case. Speaking from an efficiency perspective, the second option would be better, so I think I'll just use that. Thank you.

Reply:
@lordalcol Actually, thinking about it, that has a potential issue: It creates an uneven distribution, as `8` and `16` will occur more often, as there's more possible values in that range that will be rounded. Or do you think that is irrelevant? Also it will require this to avoid 0, which makes the code a little unclean imo.
```cpp
if (m_size.x == 0) x = 1;
if (m_size.y == 0) y = 1;
```

Reply:
So I think we can look at this from another point of view. Basically (according to your code) we want the size of the texture to be one of 2, 4, 8, 16 randomly, is that right? I think we can just do that with something like:

```
		m_size = tcu::IVec2(1 << (deRandom_getUint32(&random) % 4 + 1),
						1 << (deRandom_getUint32(&random) % 4 + 1));
```

Reply:
Ah yeah true, cause that's effectively `integerPow(2, n)`. Good point.